v2/internal/frontend/assetserver/assethandler.go

```diff
@@ -129,13 +129,14 @@ func (d *assetHandler) serveFSFile(rw http.ResponseWriter, req *http.Request, fi
 		return err
 	}
 
+	url := req.URL.Path
+	isDirectoryPath := url == "" || url[len(url)-1] == '/'
 	if statInfo.IsDir() {
-		url := req.URL.Path
-		if url != "" && url[len(url)-1] != '/' {
+		if !isDirectoryPath {
 			// If the URL doesn't end in a slash normally a http.redirect should be done, but that currently doesn't work on
-			// WebKit WebVies (macOS/Linux).
-			// So we handle this as a file that could not be found.
-			return os.ErrNotExist
+			// WebKit WebViews (macOS/Linux).
+			// So we handle this as a specific error
+			return fmt.Errorf("a directory has been requested without a trailing slash, please add a trailing slash to your request")
```

Comment:
I'm wondering if there's a consequence to just accepting this as we already know it's a directory and I think the `path.Join()` below would still work? Maybe it's best to error but thought I'd ask your thoughts on it 👍

Reply:
Yeah, from our side everything would work, but I think this would give some problems for the users. 

Let's consider the following case, someone navigates to  `/test` which is a folder and would be handled by this case. The file `/test/index.html` is found and would be served. So for the WebView the url would still be `/test`. If the `index.html` contains some relative references e.g. `<script src="main.js"></script>`, the webview would then request `/main.js` instead of `/test/main.js`. But the user expects `/test/main.js` to be loaded.

Reply:
Normally Webservers would to a HTTP redirect in this case, but we currently can't do that since WebKit on macOS and Linux doesn't support HTTP 30x redirects.